pkg/gadget-collection/gadgets/trace/dns/gadget.go

```diff
@@ -160,63 +124,27 @@ func (t *Trace) Start(trace *gadgetv1alpha1.Trace) {
 		return
 	}
 
-	eventCallback := func(key string) func(event types.Event) {
-		return func(event types.Event) {
-			t.publishEvent(trace, &event, key)
-		}
-	}
-
-	genKey := func(container *containercollection.Container) string {
-		if container.Netns == t.netnsHost {
-			return "host"
-		}
-		return container.Namespace + "/" + container.Podname
-	}
-
-	attachContainerFunc := func(container *containercollection.Container) error {
-		key := genKey(container)
+	eventCallback := func(container *containercollection.Container, event types.Event) {
+		// Enrich with data from container
+		event.Node = trace.Spec.Node
+		event.Namespace = container.Namespace
+		event.Pod = container.Podname
```

Comment:
This enrichment changes the behaviour compared to before in case of pods with `hostNetwork=true`:

In this example, the DNS query to wikipedia.org is attributed to the gadget pod. In the previous code, it was not attributed to a specific container:
```
$ kubectl gadget trace dns -n gadget
NODE           NAMESPACE     POD           QR NAMESERVER      TYPE     QTYPE    NAME           
minikube       gadget        gadget-n5lln  Q  127.0.0.11      OUTGOING A        wikipedia.org.
```

```
$ minikube ssh curl wikipedia.org
```

This could be fixed by adding a boolean `HostNetwork` in `struct Container` from `containers.go`. This field does not currently exist; it could be added next to `Netns`.

Reply:
Good idea, this simplifies the code overall.